src/main/java/com/redhat/devtools/intellij/telemetry/core/service/segment/SegmentBroker.java

```diff
@@ -0,0 +1,201 @@
+import com.redhat.devtools.intellij.telemetry.core.service.Environment;
+import com.redhat.devtools.intellij.telemetry.core.service.TelemetryEvent;
+import com.redhat.devtools.intellij.telemetry.core.service.TelemetryService;
+import com.redhat.devtools.intellij.telemetry.core.util.Lazy;
+import com.redhat.devtools.intellij.telemetry.core.util.MapBuilder;
+import com.segment.analytics.Analytics;
+import com.segment.analytics.messages.IdentifyMessage;
+import com.segment.analytics.messages.MessageBuilder;
+import com.segment.analytics.messages.PageMessage;
+import com.segment.analytics.messages.TrackMessage;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class SegmentBroker implements IMessageBroker {
+
+    private static final Logger LOGGER = Logger.getInstance(SegmentBroker.class);
+
+    public static final String PROP_NAME = "name";
+    public static final String PROP_VERSION = "version";
+    public static final String PROP_APP = "app";
+    public static final String PROP_IP = "ip";
+    public static final String PROP_COUNTRY = "country";
+    public static final String PROP_LOCALE = "locale";
+    public static final String PROP_LOCATION = "location";
+    public static final String PROP_OS = "os";
+    public static final String PROP_OS_NAME = "os_name";
+    public static final String PROP_OS_VERSION = "os_version";
+    public static final String PROP_TIMEZONE = "timezone";
+
+    public static final String VALUE_NULL_IP = "0.0.0.0"; // fixed, faked ip addr
+
+    public static final String PROP_EXTENSION_NAME = "extension_name";
+    public static final String PROP_EXTENSION_VERSION = "extension_version";
+    public static final String PROP_APP_NAME = "app_name";
+    public static final String PROP_APP_VERSION = "app_version";
+
+    private static final int FLUSH_INTERVAL = 10000;
+
+    enum Type {
+        IDENTIFY {
+            public MessageBuilder toMessage(TelemetryEvent event, Map<String, Object> context, SegmentBroker broker) {
+                return broker.toMessage(IdentifyMessage.builder(), event, context);
+            }
+        },
+        TRACK {
+            public MessageBuilder toMessage(TelemetryEvent event, Map<String, Object> context, SegmentBroker broker) {
+                return broker.toMessage(TrackMessage.builder(event.getName()), event, context);
+            }
+        },
+        PAGE {
+            public MessageBuilder toMessage(TelemetryEvent event, Map<String, Object> context, SegmentBroker broker) {
+                return broker.toMessage(PageMessage.builder(event.getName()), event, context);
+            }
+
+        };
+
+        public abstract MessageBuilder toMessage(TelemetryEvent event, Map<String, Object> context, SegmentBroker broker);
+
+        public static Type valueOf(TelemetryService.Type serviceType) {
+            switch (serviceType) {
+                case USER:
+                    return IDENTIFY;
```

Comment:
text alignment

Reply:
sorry, I dont understand. I just verified, that's how the IJ autoformatter would format the switch case. What's wrong with it?